src/routes/api/db-migrations/db-migration-utils.ts

```diff
@@ -0,0 +1,96 @@
+import { Request } from "express";
+import fs from "fs";
+import path from "path";
+import { isNodeDev, isNodeTest, isNodeProd, getNodeEnv } from "utils/is-node-env";
+import { getLogger } from "config/logger";
+import { exec as execOrigin } from "child_process";
+import { promisify } from "util";
+
+const exec = promisify(execOrigin);
+const logger = getLogger("DBMigration");
+
+/**
+ * Return the common param in req.body `targetScript` to migrate up or rollback
+ * Note: Intentially to make this param case sensitive to be more safe
+ */
+export const getTargetScript = (req: Request) => {
+	let targetScript = (req.body || {}).targetScript;
+	if (!targetScript) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript" is mandatory in the request body, but found none.`
+		};
+	}
+	//just make sure the script name matching the ones in db down the track.
+	if (!targetScript.endsWith(".js")) targetScript = targetScript + ".js";
```

Comment:
/nit I'd've renamed the method to "extractTargetScript" because this is not a plain getter anymore but getter + some logic...

Reply:
Fixed, but I ended up removing the logic here, ( and also `targetScript.toLowercase() ===` in another place as well)

So the idea is keep it safe and straightforward, no logic, just use what user input.